### PR TITLE
Adds predict stage for glue tasks, and generate result files which can be submitted to gluebenchmark.com

### LINKED ISSUE #3176: GLUE test set predictions
# 🚀 Feature request

## Motivation
The `run_glue` script is super helpful.  But it currently doesn't implement producing predictions on the test datasets for the GLUE tasks.  I think this would be extremely helpful for a lot of people.  I'm sure there are plenty of people who have implemented this functionality themselves, but I haven't found any.  Since `transformers` already provides train and dev for GLUE, it would be cool to complete the feature set with providing test set predictions.

## Your contribution

I'm personally working on a branch that extends the `glue_processors` to support the test sets (which are already downloaded by the recommended `download_glue.py` script.  I also update the `run_glue.py` script to produce the `*.tsv` files required by the GLUE online submission interface.

I think I'm a couple days out from testing/completing my implementation.  I'm also sure plenty of implementations exist of this.  If there are no other plans to support this in the works, I'm happy to submit a PR.

### PR DESCRIPTION
**Pull request by stdcoutzyx**
By simply fine-tune robterta-large with 3k steps on several tasks, achieved:

Task | Metrics | Score
-- | -- | --
Microsoft Research Paraphrase Corpus | F1 / Accuracy | 91.5/88.6
Semantic Textual Similarity Benchmark | Pearson-Spearman Corr | 90.7/90.2
Quora Question Pairs | F1 / Accuracy | 69.5/87.3
Recognizing Textual Entailment | Accuracy | 82.0
Winograd NLI | Accuracy | 65.1

### CODE REVIEW

**Comment by julien-c** [comment]
Looks good!

I just improved consistency with other scripts we have (in particular, `run_ner.py`) by:
- using an enum instead of two boolean flags
- I also always append the actual label name in the predictions file, which removes the need for a new arg

Let me know if that works for you and I'll merge to master soon

**Comment by stdcoutzyx** [comment]
Thanks Julien for the code improvement! This look very good to me.

**Comment by julien-c** [comment]
Thank you for contributing this!
